src/Compilers/Core/Portable/StrongName/DesktopStrongNameProvider.cs

```diff
@@ -386,21 +347,14 @@ private unsafe void Sign(string filePath, ImmutableArray<byte> keyPair)
             }
         }
 
-        public override bool Equals(object obj)
+        public override int GetHashCode()
         {
-            // Explicitly check that we're not comparing against a derived type
-            if (obj == null || GetType() != obj.GetType())
-            {
-                return false;
-            }
-
-            var other = (DesktopStrongNameProvider)obj;
-            return _keyFileSearchPaths.SequenceEqual(other._keyFileSearchPaths, StringComparer.Ordinal);
+            return Hash.CombineValues(_keyFileSearchPaths, StringComparer.Ordinal);
         }
```

Comment:
If `Equals` uses `ReferenceEquals`, this can probably be reduced to `return base.GetHashCode();`. Better yet, would it be a breaking change to simply remove `Equals` and `GetHashCode`?

Reply:
It is an API breaking change.